content/en/blog/_posts/2021-07-15-server-side-apply-ga.md

```diff
@@ -0,0 +1,172 @@
+---
+layout: blog
+title: "Kubernetes 1.22: Server Side Apply moves to GA"
+description: >
+  Server Side Apply moves to GA.
+date: 2021-07-15T10:00:00-08:00
+slug: Server-Side-Apply-GA
+---
+
+Authors: Jeffrey Ying, Google & Joe Betz, Google
+
+Server-side Apply (SSA) has been promoted to GA in the Kubernetes v1.22 release. Support for Server-side Apply was introduced as alpha in Kubernetes v1.14 release, promoted to beta in the Kubernetes v1.16 release, and [promoted to beta 2](https://kubernetes.io/blog/2020/04/01/kubernetes-1.18-feature-server-side-apply-beta-2/) in the Kubernetes v1.18 release.
+
+The GA milestone indicates that Kubernetes users may depend on the feature and its API without fear of backwards incompatible changes in future causing regressions. GA features are protected by the [Kubernetes deprecation policy](https://kubernetes.io/docs/reference/using-api/deprecation-policy/).
+
+What is Server-side Apply?
+
+Server-side Apply helps users and controllers manage their resources through declarative configurations. Server-side Apply replaces the client side apply feature implemented by “kubectl apply” with a server-side implementation, permitting use by tools/clients other than kubectl. To learn more about why Server-side Apply is important, or to learn how to use Server-side Apply from `kubectl`, see the [Beta 2 release announcement](https://kubernetes.io/blog/2020/04/01/kubernetes-1.18-feature-server-side-apply-beta-2/).
+
+What’s new since Beta?
+
+Since the [Beta 2 release](https://kubernetes.io/blog/2020/04/01/kubernetes-1.18-feature-server-side-apply-beta-2/) subresources support has been added, and both client-go and Kubebuilder have added comprehensive support for Server-side Apply. This completes the Server-side Apply functionality required to make controller development practical.
+Subresource Support
+Server-side Apply now fully supports subresources like status and scale. This is particularly important for controllers, which are often responsible for writing to subresources.
```

Comment:
```suggestion
Server-side Apply now fully supports subresources like `status` and `scale`. This is particularly important for [controllers](/docs/concepts/architecture/controller/), which are often responsible for writing to subresources.
```

Reply:
Maybe an example, eg:
```markdown

In the API, if you have a StatefulSet named `example` you can fetch it as 
`/apis/apps/v1/namespaces/default/statefulsets/example`.
Kubernetes subresources allow fine-grained access control, so that you can allow
acccess to write to `/apis/apps/v1/namespaces/default/statefulsets/example/scale` but
not to change any other field. Now, the controller that makes that update can rely on
server-side apply to help manage potential conflicts.
```